nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java

```diff
@@ -736,24 +735,22 @@ private void executeDML(ProcessContext context, ProcessSession session, FlowFile
             final Integer maxBatchSize = context.getProperty(MAX_BATCH_SIZE).evaluateAttributeExpressions(flowFile).asInteger();
             int currentBatchSize = 0;
             int batchIndex = 0;
-
+            List<List<Object>> valuesLists = new ArrayList<>();
```

Comment:
Populating `valueLists` will involve keeping every field value from every record in memory. For flowfiles with a large number of records, this could blow the heap and cause an OutOfMemory error

Reply:
Yes, you are absolutely right！ I have modified it